src/rgw/rgw_file.h

```diff
@@ -504,6 +504,14 @@ namespace rgw {
       return nullptr;
     }
 
+    int offset_of(const std::string& name, int64_t *offset, uint32_t flags) {
+      if (unlikely(! is_dir())) {
+	return -EINVAL;
+      }
+      *offset = XXH64(name.c_str(), name.length(), fh_key::seed);
```

Comment:
nit: since this doesn't depend on a null-terminated string, consider taking the string as a `string_view` to avoid constructing a `std::string`